app/src/main/java/de/moyapro/idleworldsim/domain/Biome.kt

```diff
@@ -23,16 +30,63 @@ data class Biome(
 
     fun process(): Biome {
         this.resources += generation
-        this.resources = speciesList
-            .shuffled()
-            .applyTo(resources, Species::process)
+//        calculate Map<Species, Map<ResourceType,  AquireSkill>> to distribute available resources
+//        determin Map<Species, Resources> how much is the species able to consume
+
+        val availableResourcePerSpecies: Map<Species, Resources> = getAquiredResourcesPerSpecies()
+        speciesList
+            .forEach {
+                val available = availableResourcePerSpecies[it] ?: emptyResources()
+                available.populations = resources.populations
+                val leftovers = it.process(available)
+                val used = available - leftovers
+                resources -= used
+            }
         onBiomeProcess.notifyChange()
         return this
     }
 
-    fun settle(species: Species): Biome {
+    fun getAquiredResourcesPerSpecies(): Map<Species, Resources> {
+        val resultMap = resources.getSpecies().associateWithTo(mutableMapOf()) { emptyResources() }
+        for (resourceType in values()) {
+            val needPerSpecies: Map<Species, Resource> = resources.populations
+                .filter { it.key.hasTrait(ConsumerTrait(resourceType)) && it.key.hasTrait(NeedResource(resourceType)) }
+                .map { (species, population) -> Pair(species, species.needsPerIndividual()[resourceType] * population) }
+                .associate { it }
+            val totalAvailable: Resource = resources[resourceType]
+            val totalNeed = Resource(resourceType, needPerSpecies.values.sumByDouble { it.amount })
+            if (totalNeed <= totalAvailable) {
+                distributeWithSurplus(resultMap, needPerSpecies)
+            } else {
+                distributeWithShortage(resultMap, resourceType, totalAvailable)
+            }
+        }
+        return resultMap
+    }
+
+    private fun distributeWithShortage(resultMap: MutableMap<Species, Resources>, resourceType: ResourceType, totalAvailable: Resource) {
+        if (totalAvailable.isNone()) {
+            return
+        }
+        val maxAquireSkill = max(1, resources.populations.keys.map { it[AquireResource(resourceType)].level }.max() ?: 1)
```

Comment:
Wir hatten ja schon darüber diskutiert, ob immer alle zur Verfügung stehenden Ressourcen aufgeteilt werden. Ich finde das nicht sinnvoll. Wenn ein Überfluss existiert, kann jede Spezies den Bedarf decken, lässt aber viel für andere übrig. Dann gibt es keine Konkurrenz. Mein Gefühl sagt mir, dass sollten wir auch abbilden können.
Ich stelle es mir so vor, dass erst eine Verteilung ausgerechnet wird, die schrittweise vom höchsten zum niedrigsten Level geht. Gestartet wird mit deiner hier angenommenen Verteilung und die Spezies mit dem höchsten Level bedient sich als erste, allerdings nur bis max. dem Anteil, der zur Verfügung steht. Anschließend werden nicht benötigte Ressourcen als Übertrag zur nächsten Spezies genommen. Nachteil des Verfahrens: Mehr Rechenaufwand. Vorteil: differenziertere Effekte.

Reply:
PS: Du kannst den Punkt aber gerne abhaken. Sowas lässt sich auch später einbauen.

Reply:
Aktuell ist es ein zweiteiliger Prozess. In dem Codeteil oben wird eine Obergrenze berechnet die grob einer 1/Skilldifferenz entspricht. Anschliessend gehen diese Resourcen in den "alten" process().
Wie bereits angedeutet sieht man aktueller Plan so aus, dass es ResourceProducer und ResourceConsumer geben soll. Das kann also auch das Biome genauso wie jede Spezies sein. 
Idee fuer die Umsetzung waere es einen gerichteten, gewichteten Graphen aufzubauen. Eine Kante existiert immer zwischen n Producern und m Consumern. Entlang einer Kante werden Resourcen von den Producern zu den Consumern uebertragen. Dieser Vorgang kann Effekte auf Consumer und Producer haben. Jeder Kante hat mehrere Gewichtungen. Wie viele Resourcen sind fuer den Consumer zu holen, wie gut kann sich ein Consumer gegen die Konkurenz durchsetzen und ggf was uns noch so dazu einfaellt. Die Gewichtungen ergeben sich aus den Skills der Consumer und des Producers. 
Beim Processschritt entscheidet eine Gewichtung ueber die Reihenfolge in der Gefressen wird und eine andere darueber wie viel Resourcen jeweils erhalten werden.

Reply:
P.S. Wie dokumentiert man sowas? GG

Reply:
Idee: Wenn man auf nichtlineare Zusammenhänge (z.B. Fallunterscheidungen) verzichten kann/möchte, könnte auch eine Übergangsmatrix helfen. Diese ließe sich effizient jedes Mal anwenden ohne komplexe Algorithmen. Die Übergangsmatrix würde jedes Mal neu berechnet werden müssen, wenn sich die Bedingungen verändern. Für die Bestimmung der Matrix kann z.B. der o.g. Graph als Ausgangspunkt dienen. Wie gesagt... nur lineare Abhängigkeiten vorausgesetzt.